src/pip/_internal/req/constructors.py

```diff
@@ -80,7 +80,8 @@ def _set_requirement_extras(req: Requirement, new_extras: Set[str]) -> Requireme
     assert (
         pre is not None and post is not None
     ), f"regex group selection for requirement {req} failed, this should never happen"
-    extras: str = "[%s]" % ",".join(sorted(new_extras)) if new_extras else ""
+    extras: str = ",".join(sorted(new_extras)) if new_extras else ""
+    extras = f"[{extras}]"
```

Comment:
Is this the only way to write this under the upgrade rules, is there not some one-liner that's acceptable? e.g.

```python
extras = f"[{','.join(sorted(new_extras))}]" if new_extras else ""
```

Reply:
Sure. I found it less readable, but am happy to change to a singe line if you prefer it..

Reply:
Personally I find parsing across two lines less readable, but I do agree my suggestion is less readable than the original code, so maybe just put a noqa.

Reply:
Changed to `.format()` instead of f-string as suggested by the rule.